File: solutions/elixir/guessing-game/1/lib/guessing_game.ex
defmodule GuessingGame do
  def compare(secret_number, guess \\ :no_guess) do
    cond do
      guess == :no_guess -> "Make a guess"
      guess in [secret_number + 1, secret_number - 1] -> "So close"
      guess > secret_number -> "Too high"
      guess < secret_number -> "Too low"
      true -> "Correct"
    end
  end
end

File: solutions/haskell/guessing-game/1/src/GuessingGame.hs
module GuessingGame (reply) where

reply :: Int -> String
reply 42 = "Correct"
reply 41 = "So close"
reply 43 = "So close"
reply guess
  | guess < 41 = "Too low"
  | guess > 43 = "Too high"
reply _ = error "huh?"
